idaes/gas_solid_contactors/properties/methane_iron_OC_reduction/gas_phase_thermo.py

```diff
@@ -289,9 +290,12 @@ def initialize(blk, state_args=None, hold_state=False,
                                     "for state block is not zero during "
                                     "initialization.")
 
-        # Set solver options
-        opt = SolverFactory(solver)
-        opt.options = optarg
+        # Create solver
+        if solver is None:
+            opt = get_default_solver()
+        else:
+            opt = SolverFactory(solver)
+            opt.options = optarg
```

Comment:
This one's going to be problematic.  Since default solver comes with some options set and optarg is an empty dict, this will overwrite the default settings.  I think you want ```opt.options.update(optarg)```

Reply:
I hate to suggest this given how many times this if/else block appears, but with PR #207, I think you can replace this (assuming that PR gets approved) with:

```python
from idaes.core.solvers import SolverFactory
...
opt = SolverFactory(solver, options=optarg)
```

If solver is ```None```, you'll get the default solver specified by the IDAES config, which by default is ipopt with the ```get_default_solver()``` options.

Reply:
I had been thinking of ways we could eliminate all the if/else statements, as it is a lot of repeated code. If PR #207 will fix that, then that is a good thing.